src/main/business/MetaQuestions/MetaQuestionController.js

```diff
@@ -1,83 +1,121 @@
+            keywords: [PRIMITIVE_TYPES.STRING],
+            tag: PRIMITIVE_TYPES.STRING,
+            title: PRIMITIVE_TYPES.STRING,
+            content: PRIMITIVE_TYPES.STRING,
+        });
+
+        const dalAppendix = await this.#metaQuestionRepo.addAppendix(data, data.keywords);
+        return new Appendix(dalAppendix);
+    }
+
+    /**
+     * Updates a given appendix with new data (keywords, title, content)
+     * @param data
+     * @return {Promise<Appendix>} The updated appendix
+     */
+    async editAppendix(data) {
+        validateParameters(data, {
+            keywords: [PRIMITIVE_TYPES.STRING],
+            tag: PRIMITIVE_TYPES.STRING,
+        });
+
+        const dalAppendix = await this.#metaQuestionRepo.setKeywordsToAppendix(data.tag, data.keywords)
+        const appendix = new Appendix(dalAppendix);
+        if (data.title) {
+            await appendix.setTitle(data.title);
         }
-        let metaQuestion = new MetaQuestion(data);
-        this.#metaQuestions.set(this.#metaQuestionId, metaQuestion);
-        const ta_s = this.#userController.getAllStaff(data)["TAs"]
+        if (data.content) {
+            await appendix.setContent(data.content);
+        }
+        return appendix;
+    }
 
+    async createMetaQuestion(data) {
+        validateParameters(data, {
+            keywords: [PRIMITIVE_TYPES.STRING],
+            answers: [{content: PRIMITIVE_TYPES.STRING, tag: PRIMITIVE_TYPES.STRING}],
+            stem: PRIMITIVE_TYPES.STRING,
+        });
+
+        const dalMQ = await this.#metaQuestionRepo.addMetaQuestion(data, data.answers, data.keywords);
+
+        let metaQuestion = new MetaQuestion(dalMQ);
+        const ta_s = await this.#userController.getAllStaff(data)["TAs"]
         const addTaskProperties = {...data,
              assignedUsers: ta_s, taskType: TaskTypes.ADD_KEY,
               taskPriority: TaskPriority.HIGH, description: `Please add a key For the following Question: ${metaQuestion.stem}`,
             metaQuestion: metaQuestion}
         this.#taskController.addTask(addTaskProperties)
-        this.#metaQuestionId = this.#metaQuestionId + 1
         return metaQuestion
     }
 
-    editMetaQuestion(data) {
-        let metaQuestion = new MetaQuestion(data);
-        this.#metaQuestions.set(data.id, metaQuestion);
+    async editMetaQuestion(data) {
+        validateParameters(data, {
+            id: PRIMITIVE_TYPES.NUMBER,
+            keywords: [PRIMITIVE_TYPES.STRING],
+            answers: [{content: PRIMITIVE_TYPES.STRING, tag: PRIMITIVE_TYPES.STRING}],
+            stem: PRIMITIVE_TYPES.STRING,
+        });
+        await this.#metaQuestionRepo.setKeywordsToQuestion(data.id, data.keywords);
+        await this.#metaQuestionRepo.deleteAnswersOfMq(data.id)
+        await this.#metaQuestionRepo.addAnswersToQuestion(data.id, data.answers);
+        const metaQuestion = await this.getMetaQuestion(data.id);
+        await metaQuestion.setStem(data.stem);
```

Comment:
4 awaits for the db for a single edit ? Cant you make a function that do it in one ?
If not, wouldnt be better to do Promise.all() and let them run asycronesly ?

Reply:
Ye I don't like it as well, but this is just to match the existing logic (which replaces the whole mq with the given data)
This should be done smoother. 
using `Promise.all` isn't possible due to order that should be kept (with the answers)

Reply:
I dont know what order should be kept, but I believe you...

But think if you can make it more asynchronously (maybe use then on 2 promises, and the other will be asynchronously)

Reply:
Sounds like a good idea

Reply:
After re-reviewing the situation I prefer to leave the code as is. It is correct and refactoring it into a 'better looking' code that stays correct is way too hard. 
I'll fix it in another chance to come.